lib/t/accounts.ex

```diff
@@ -458,11 +469,17 @@ defmodule T.Accounts do
     end)
     |> Repo.transaction()
     |> case do
-      {:ok, %{user: user, profile: %Profile{} = profile}} ->
-        m = "user registered #{user.id}, user name is #{profile.name}"
-        Logger.warn(m)
-        Bot.async_post_message(m)
-        {:ok, %Profile{profile | hidden?: false}}
+      {:ok, changes} ->
+        case changes do
+          %{user: user, profile: %Profile{} = profile, gender_preferences: genders} ->
+            m = "user registered #{user.id}, user name is #{profile.name}"
+            Logger.warn(m)
+            Bot.async_post_message(m)
+            {:ok, %Profile{profile | gender_preference: genders, hidden?: false}}
+
+          %{profile: profile} ->
+            {:ok, %Profile{profile | hidden?: false}}
```

Comment:
why is this clause necessary?

Reply:
because there may be no :gender_preferences in this case: https://github.com/getsince/test3/blob/b94aebd7d5d4c04b4ef98b3054e7eaaccd8c9f01/lib/t/accounts.ex#L553

Reply:
please use two clauses, current approach binds notification logic to ui, hard to understand